source/game/kart/KartMove.cc

```diff
@@ -26,14 +56,23 @@ void KartMove::setInitialPhysicsValues(const EGG::Vector3f &pos, const EGG::Vect
             EGG::Vector3f::inf, KCL_ANY, &info, &kcl_flags, 0);
 
     if (bColliding && (kcl_flags & KCL_TYPE_FLOOR)) {
-        newPos += info.m_minPlusMax + (info.m_floorNrm * -100.0f);
-        newPos += info.m_floorNrm * bsp().m_initialYPos;
+        newPos += info.tangentOff + (info.floorNrm * -100.0f);
+        newPos += info.floorNrm * bsp().initialYPos;
     }
 
     setPos(newPos);
     setRot(quaternion);
 
     sub()->initPhysicsValues();
+
+    physics()->setPos(pos());
+    physics()->setVelocity(dynamics()->velocity());
+
+    m_dir = bodyFront();
+
+    for (u16 tireIdx = 0; tireIdx < suspCount(); ++tireIdx) {
+        suspension(tireIdx)->setInitialState();
+    }
```

Comment:
So with these loops, maybe its worth having extra proxy functions that return an array, so we can do `for (auto &x : y)`. Thoughts?